backend/backend.go

```diff
@@ -102,6 +102,18 @@ type Backend interface {
 	// PersistState is called, depending on the state manager implementation.
 	StateMgr(workspace string) (statemgr.Full, error)
 
+	// StateMgrWithoutCheckVersion returns the state manager for the given
+	// workspace name, while ensuring that Terraform version checks are not
+	// performed if the backend needs to read a state file in order to
+	// initialize the state manager.
+	//
+	// For backends which do not need to read a state file at this point, this
+	// is identical to StateMgr.
+	//
+	// This is used to facilitate reading compatible state files from newer
+	// versions of Terraform.
+	StateMgrWithoutCheckVersion(workspace string) (statemgr.Full, error)
```

Comment:
I considered this but went with a separate method for a couple of reasons:

- I have a subjective dislike of adding booleans to the end of functions, because I find the call sites difficult to read and it's easy to get the boolean inverted. `DoSomething(foo, true)` is less clear to me than `DoSomethingAndVerify(foo)`
- I think it makes for a smaller diff this way, too, at least for the commit which has an effect. I don't have to touch all the `StateMgr` call sites, only the Terraform provider one